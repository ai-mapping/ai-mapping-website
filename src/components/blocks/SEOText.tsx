import Link from 'next/link';
import React from 'react';
import Section from '@/components/layout/Section';

// Define types for model links
type ModelLink = {
  href: string;
  label: string;
};

// Define types for brand sections
type BrandSection = {
  title: string;
  models: ModelLink[];
};

// Main props interface
export type SEOTextProps = {
  heading: string;
  paragraphs: string[];
  brandSections: BrandSection[];
  footerText: string;
};

// Default props
export const defaultSEOTextProps: SEOTextProps = {
  heading: 'Запчасти для техники Caterpillar и KOMATSU',
  paragraphs: [
    'В условиях санкций и ухода официальных дилеров с российского рынка, поиск качественных запчастей для спецтехники Caterpillar и KOMATSU стал серьезной проблемой для многих компаний. Простой дорогостоящей техники приводит к значительным финансовым потерям и срывам сроков выполнения работ.',
    'Наша компания предлагает эффективное решение этой проблемы – экспресс-доставку запчастей из Китая в сжатые сроки. Мы специализируемся на поставках качественных аналогов для всех моделей техники Caterpillar и KOMATSU, а также других ведущих производителей.',
    'Благодаря налаженным связям с производителями в Китае и собственной логистической сети, мы обеспечиваем поставку запчастей в Москву и Московскую область в течение 7-14 дней, что в разы быстрее, чем у большинства конкурентов.'
  ],
  brandSections: [
    {
      title: 'Популярные модели техники Caterpillar:',
      models: [
        { href: '#', label: 'CAT 320D' },
        { href: '#', label: 'CAT 330D' },
        { href: '#', label: 'CAT 336D' },
        { href: '#', label: 'CAT 345D' },
        { href: '#', label: 'CAT D6R' },
        { href: '#', label: 'CAT D9R' },
        { href: '#', label: 'CAT 950H' },
        { href: '#', label: 'CAT 966H' }
      ]
    },
    {
      title: 'Популярные модели техники KOMATSU:',
      models: [
        { href: '#', label: 'KOMATSU PC200' },
        { href: '#', label: 'KOMATSU PC300' },
        { href: '#', label: 'KOMATSU PC400' },
        { href: '#', label: 'KOMATSU D65' },
        { href: '#', label: 'KOMATSU D155' },
        { href: '#', label: 'KOMATSU WA470' },
        { href: '#', label: 'KOMATSU WA500' },
        { href: '#', label: 'KOMATSU HD785' }
      ]
    }
  ],
  footerText: 'Мы работаем как с частными компаниями, так и с крупными строительными корпорациями в Москве и Московской области. При необходимости организуем доставку в другие регионы России.'
};

const SEOText = ({
  heading = defaultSEOTextProps.heading,
  paragraphs = defaultSEOTextProps.paragraphs,
  brandSections = defaultSEOTextProps.brandSections,
  footerText = defaultSEOTextProps.footerText
}: Partial<SEOTextProps>) => {
  return (
    <Section background="background" paddingY="lg" containerWidth="medium">
          <h2 className="text-2xl font-semibold mb-4">
            {heading}
          </h2>

          <div className="prose prose-lg max-w-none text-dark mb-6">
            {paragraphs.map((paragraph, index) => (
              <p key={`paragraph-${index}`}>{paragraph}</p>
            ))}
          </div>

          {brandSections.map((section, sectionIndex) => (
            <React.Fragment key={`section-${sectionIndex}`}>
              <h3 className="text-xl font-semibold mb-3">
                {section.title}
              </h3>
              <div className="grid grid-cols-2 md:grid-cols-4 gap-2 mb-6">
                {section.models.map((model, modelIndex) => (
                  <Link 
                    key={`model-${sectionIndex}-${modelIndex}`} 
                    href={model.href} 
                    className="text-accent hover:underline"
                  >
                    {model.label}
                  </Link>
                ))}
              </div>
            </React.Fragment>
          ))}

          <p className="text-dark">
            {footerText}
          </p>
    </Section>
  );
};

export default SEOText;
