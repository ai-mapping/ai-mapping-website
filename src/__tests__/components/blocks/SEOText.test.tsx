import React from 'react';
import { render, screen } from '@testing-library/react';
import SEOText, { defaultSEOTextProps } from '@/components/blocks/SEOText';

describe('SEOText Component', () => {
  it('renders with default props correctly', () => {
    const { container } = render(<SEOText />);
    
    // Check if heading is rendered
    expect(screen.getByText(defaultSEOTextProps.heading)).toBeInTheDocument();
    
    // Check if paragraphs are rendered
    defaultSEOTextProps.paragraphs.forEach((paragraph) => {
      expect(screen.getByText(paragraph)).toBeInTheDocument();
    });
    
    // Check if brand sections are rendered
    defaultSEOTextProps.brandSections.forEach((section) => {
      expect(screen.getByText(section.title)).toBeInTheDocument();
      
      // Check if model links are rendered
      section.models.forEach((model) => {
        const link = screen.getByText(model.label);
        expect(link).toBeInTheDocument();
        expect(link.closest('a')).toHaveAttribute('href', model.href);
      });
    });
    
    // Check if footer text is rendered
    expect(screen.getByText(defaultSEOTextProps.footerText)).toBeInTheDocument();
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom props correctly', () => {
    const customProps = {
      heading: 'Custom SEO Heading',
      paragraphs: [
        'Custom paragraph 1',
        'Custom paragraph 2'
      ],
      brandSections: [
        {
          title: 'Custom Brand Section',
          models: [
            { href: '/custom-link', label: 'Custom Model 1' },
            { href: '/custom-link-2', label: 'Custom Model 2' }
          ]
        }
      ],
      footerText: 'Custom footer text'
    };
    
    const { container } = render(<SEOText {...customProps} />);
    
    // Check if custom heading is rendered
    expect(screen.getByText(customProps.heading)).toBeInTheDocument();
    
    // Check if custom paragraphs are rendered
    customProps.paragraphs.forEach((paragraph) => {
      expect(screen.getByText(paragraph)).toBeInTheDocument();
    });
    
    // Check if custom brand section is rendered
    expect(screen.getByText(customProps.brandSections[0].title)).toBeInTheDocument();
    
    // Check if custom model links are rendered
    customProps.brandSections[0].models.forEach((model) => {
      const link = screen.getByText(model.label);
      expect(link).toBeInTheDocument();
      expect(link.closest('a')).toHaveAttribute('href', model.href);
    });
    
    // Check if custom footer text is rendered
    expect(screen.getByText(customProps.footerText)).toBeInTheDocument();
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});