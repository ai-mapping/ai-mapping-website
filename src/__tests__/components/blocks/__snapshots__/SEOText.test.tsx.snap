// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEOText Component renders with custom props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-4xl"
    >
      <h2
        class="text-2xl font-semibold mb-4"
      >
        Custom SEO Heading
      </h2>
      <div
        class="prose prose-lg max-w-none text-dark mb-6"
      >
        <p>
          Custom paragraph 1
        </p>
        <p>
          Custom paragraph 2
        </p>
      </div>
      <h3
        class="text-xl font-semibold mb-3"
      >
        Custom Brand Section
      </h3>
      <div
        class="grid grid-cols-2 md:grid-cols-4 gap-2 mb-6"
      >
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="/custom-link"
        >
          Custom Model 1
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="/custom-link-2"
        >
          Custom Model 2
        </a>
      </div>
      <p
        class="text-dark"
      >
        Custom footer text
      </p>
    </div>
  </section>
</div>
`;

exports[`SEOText Component renders with default props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-4xl"
    >
      <h2
        class="text-2xl font-semibold mb-4"
      >
        Запчасти для техники Caterpillar и KOMATSU
      </h2>
      <div
        class="prose prose-lg max-w-none text-dark mb-6"
      >
        <p>
          В условиях санкций и ухода официальных дилеров с российского рынка, поиск качественных запчастей для спецтехники Caterpillar и KOMATSU стал серьезной проблемой для многих компаний. Простой дорогостоящей техники приводит к значительным финансовым потерям и срывам сроков выполнения работ.
        </p>
        <p>
          Наша компания предлагает эффективное решение этой проблемы – экспресс-доставку запчастей из Китая в сжатые сроки. Мы специализируемся на поставках качественных аналогов для всех моделей техники Caterpillar и KOMATSU, а также других ведущих производителей.
        </p>
        <p>
          Благодаря налаженным связям с производителями в Китае и собственной логистической сети, мы обеспечиваем поставку запчастей в Москву и Московскую область в течение 7-14 дней, что в разы быстрее, чем у большинства конкурентов.
        </p>
      </div>
      <h3
        class="text-xl font-semibold mb-3"
      >
        Популярные модели техники Caterpillar:
      </h3>
      <div
        class="grid grid-cols-2 md:grid-cols-4 gap-2 mb-6"
      >
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT 320D
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT 330D
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT 336D
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT 345D
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT D6R
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT D9R
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT 950H
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          CAT 966H
        </a>
      </div>
      <h3
        class="text-xl font-semibold mb-3"
      >
        Популярные модели техники KOMATSU:
      </h3>
      <div
        class="grid grid-cols-2 md:grid-cols-4 gap-2 mb-6"
      >
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU PC200
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU PC300
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU PC400
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU D65
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU D155
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU WA470
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU WA500
        </a>
        <a
          class="text-accent hover:underline"
          data-testid="next-link"
          href="#"
        >
          KOMATSU HD785
        </a>
      </div>
      <p
        class="text-dark"
      >
        Мы работаем как с частными компаниями, так и с крупными строительными корпорациями в Москве и Московской области. При необходимости организуем доставку в другие регионы России.
      </p>
    </div>
  </section>
</div>
`;
